File: jscomp/test/fs_test.js
'use strict';

Mt = require("./mt.js");
Fs = require("fs");
Path = require("path");
Block = require("../../lib/js/block.js");

suites = do
  contents: --[ [] ]--0
end;

test_id = do
  contents: 0
end;

function eq(loc, param) do
  y = param[1];
  x = param[0];
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = --[ :: ]--[
    --[ tuple ]--[
      loc .. (" id " .. String(test_id.contents)),
      (function (param) do
          return --[ Eq ]--Block.__(0, [
                    x,
                    y
                  ]);
        end end)
    ],
    suites.contents
  ];
  return --[ () ]--0;
end end

match = typeof __filename == "undefined" and undefined or __filename;

current_file = match ~= undefined and match or "<Not Node JS>";

match$1 = typeof __dirname == "undefined" and undefined or __dirname;

current_dir_name = match$1 ~= undefined and match$1 or "<Not Node Js>";

Fs.readFileSync(current_file, "utf8");

Fs.readdirSync(current_dir_name);

pathobj = Path.parse(current_dir_name);

match$2 = typeof module == "undefined" and undefined or module;

if (match$2 ~= undefined) then do
  console.log(--[ tuple ]--[
        match$2.id,
        match$2.paths
      ]);
  eq("File \"fs_test.ml\", line 45, characters 7-14", --[ tuple ]--[
        pathobj.name,
        "test"
      ]);
end
 end 

Mt.from_pair_suites("Fs_test", suites.contents);

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
--[ match Not a pure module ]--

File: jscomp/test/fs_test.lua
__console = {log = print};

Mt = require "..mt";
Fs = require "fs";
Path = require "path";
Block = require "......lib.js.block";

suites = {
  contents = --[[ [] ]]0
};

test_id = {
  contents = 0
};

function eq(loc, param) do
  y = param[2];
  x = param[1];
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = --[[ :: ]]{
    --[[ tuple ]]{
      loc .. (" id " .. __String(test_id.contents)),
      (function(param) do
          return --[[ Eq ]]Block.__(0, {
                    x,
                    y
                  });
        end end)
    },
    suites.contents
  };
  return --[[ () ]]0;
end end

match = type(__filename) == "undefined" and nil or __filename;

current_file = match ~= nil and match or "<Not Node JS>";

match_1 = type(__dirname) == "undefined" and nil or __dirname;

current_dir_name = match_1 ~= nil and match_1 or "<Not Node Js>";

Fs.readFileSync(current_file, "utf8");

Fs.readdirSync(current_dir_name);

pathobj = Path.parse(current_dir_name);

match_2 = type(module) == "undefined" and nil or module;

if (match_2 ~= nil) then do
  __console.log(--[[ tuple ]]{
        match_2.id,
        match_2.paths
      });
  eq("File \"fs_test.ml\", line 45, characters 7-14", --[[ tuple ]]{
        pathobj.name,
        "test"
      });
end
 end 

Mt.from_pair_suites("Fs_test", suites.contents);

exports = {};
exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
return exports;
--[[ match Not a pure module ]]
